File: src/hamming-distance.go
package main

func hammingDistance(x int, y int) int {
	var result = 0
	for !(x == 0 && y == 0) {
		if (x % 2) != (y % 2) {
			result++
		}
		x = x / 2
		y = y / 2
	}
	return result
}

File: src/hamming-distance.rs
impl Solution {
    pub fn hamming_distance(x: i32, y: i32) -> i32 {
        let mut n: i32 = 0;
        let mut i = 0;
        let mut a = x;
        let mut b = y;
        while (i < 32) {
            if((a%2)!=(b%2)) {
                n = n + 1 ;
            }
            a = a / 2;
            b = b / 2;
            i = i + 1;
        }
        return n
    }
}
